server/src/main/java/org/elasticsearch/action/admin/indices/resolve/TransportResolveClusterAction.java

```diff
@@ -0,0 +1,286 @@
+
+                ActionListener<ResolveClusterActionResponse> remoteListener = new ActionListener<>() {
+                    @Override
+                    public void onResponse(ResolveClusterActionResponse response) {
+                        if (resolveClusterTask.isCancelled()) {
+                            releaseResourcesOnCancel.run();
+                            return;
+                        }
+                        ResolveClusterInfo info = response.getResolveClusterInfo().get(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+                        if (info != null) {
+                            clusterInfoMap.put(
+                                clusterAlias,
+                                new ResolveClusterInfo(info.isConnected(), skipUnavailable, info.getMatchingIndices(), info.getBuild())
+                            );
+                        }
+                        if (resolveClusterTask.isCancelled()) {
+                            releaseResourcesOnCancel.run();
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Exception failure) {
+                        if (resolveClusterTask.isCancelled()) {
+                            releaseResourcesOnCancel.run();
+                            return;
+                        }
+                        System.err.println(">>> HHHHH failure FAILURE: " + failure.getClass() + "; " + failure);
+                        if (notConnectedError(failure)) {
+                            clusterInfoMap.put(clusterAlias, new ResolveClusterInfo(false, skipUnavailable));
+                        } else if (ExceptionsHelper.unwrap(
+                            failure,
+                            ElasticsearchSecurityException.class
+                        ) instanceof ElasticsearchSecurityException ese) {
+                            clusterInfoMap.put(clusterAlias, new ResolveClusterInfo(true, skipUnavailable, ese.getMessage()));
+                        } else if (ExceptionsHelper.unwrap(failure, IndexNotFoundException.class) instanceof IndexNotFoundException infe) {
+                            System.err.println(">>>> HHH INDEX NOT FOUND!!! <<<<<<<<<<<<<<<");
+                            clusterInfoMap.put(clusterAlias, new ResolveClusterInfo(true, skipUnavailable, infe.getMessage()));
+                        } else {
+                            Throwable cause = ExceptionsHelper.unwrapCause(failure);
+                            // when querying an older cluster that does not have the _resolve/cluster endpoint,
+                            // we will get this error at the Transport layer, since there are version guards
+                            // on the Writeables for this Action.
+                            if (cause instanceof UnsupportedOperationException
+                                && cause.getMessage().contains("ResolveClusterAction requires at least Transport Version")) {
```

Comment:
nit: put this text in a static string incase someone decides to change the wording.

Reply:
Fixed.